desmume/src/frontend/posix/gtk/main.cpp

```diff
@@ -1779,6 +1795,115 @@ static void PlayMovieDialog(GSimpleAction *action, GVariant *parameter, gpointer
     g_object_unref(pFileSelection);
 }
 
+static void ImportBackupMemoryDialog(GSimpleAction *action, GVariant *parameter, gpointer user_data)
+{
+    GtkFileFilter *pFilter_raw, *pFilter_ar, *pFilter_any;
+    GtkFileChooserNative *pFileSelection;
+
+    if (desmume_running())
+        Pause(NULL, NULL, NULL);
+
+    pFilter_raw = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_raw, "*.sav");
+    gtk_file_filter_set_name(pFilter_raw, "Raw/No$GBA Save format (*.sav)");
+
+    pFilter_ar = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_ar, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_ar, "*.dss");
+    gtk_file_filter_set_name(pFilter_ar, "Action Replay DS Save (*.duc,*.dss)");
+
+    pFilter_any = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(pFilter_any, "*.sav");
+    gtk_file_filter_add_pattern(pFilter_any, "*.duc");
+    gtk_file_filter_add_pattern(pFilter_any, "*.dss");
+    gtk_file_filter_set_name(pFilter_any, "All supported types");
+
+    /* Creating the selection window */
+    pFileSelection = gtk_file_chooser_native_new("Import Backup Memory From ...",
+            GTK_WINDOW(pWindow),
+            GTK_FILE_CHOOSER_ACTION_OPEN,
+            "_Open", "_Cancel");
+
+    /* Only the dialog window is accepting events: */
+    gtk_window_set_modal(GTK_WINDOW(pFileSelection), TRUE);
```

Comment:
You can’t cast a `GtkFileChooserNative` into a `GtkWindow`, it isn’t even necessarily a `GtkWidget` as it could be implemented by a completely foreign toolkit, possibly even out-of-process.

This prints a critical warning when running it.

Reply:
Thanks; I've removed this in e036298aa2dc0903bdd2417f49f1de7e35cfea5f